pkg/mimirtool/client/client.go

```diff
@@ -97,7 +98,10 @@ func New(cfg Config) (*MimirClient, error) {
 
 	path := rulerAPIPath
 	if cfg.UseLegacyRoutes {
-		path = legacyAPIPath
+		var err error
+		if path, err = url.JoinPath(cfg.MimirHTTPPrefix, legacyAPIPath); err != nil {
+			return nil, err
+		}
 	}
```

Comment:
Should we validate that the user mustn't set `MimirHTTPPrefix` if `UseLegacyRoutes` is false?

Reply:
this could be tricky to check here, since `MimirHTTPPrefix` has a default value /prometheus, in client we will not able to distinguish the not set and default value.